ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/streaming/WSQueryEndpointTest.java

```diff
@@ -133,17 +143,25 @@ public void setUp() {
         Optional.empty(), Optional.empty(), Optional.empty(), Optional.empty(), OptionalInt.empty()
     );
     when(session.getId()).thenReturn("session-id");
+    when(session.getUserPrincipal()).thenReturn(principal);
     when(statementParser.parseSingleStatement(anyString()))
         .thenAnswer(invocation -> PreparedStatement.of(invocation.getArgument(0).toString(), query));
-    when(serviceContext.getSchemaRegistryClient()).thenReturn(schemaRegistryClient);
-    when(serviceContext.getTopicClient()).thenReturn(topicClient);
+    when(securityExtension.getSchemaRegistryClientSupplier(principal)).thenReturn(() -> schemaRegistryClient);
+    when(securityExtension.getKafkaClientSupplier(principal)).thenReturn(topicClientSupplier);
+    when(topicClientSupplier.getAdminClient(any())).thenReturn(adminClient);
     when(ksqlEngine.isAcceptingStatements()).thenReturn(true);
     givenRequest(VALID_REQUEST);
 
     wsQueryEndpoint = new WSQueryEndpoint(
-        ksqlConfig, OBJECT_MAPPER, statementParser, ksqlEngine, serviceContext, commandQueue, exec,
+        ksqlConfig, OBJECT_MAPPER, statementParser, ksqlEngine, commandQueue, exec,
         queryPublisher, topicPublisher, activenessRegistrar, COMMAND_QUEUE_CATCHUP_TIMEOUT,
-        topicAccessValidator);
+        topicAccessValidator, securityExtension);
+  }
+
+  @Test
+  public void shouldInitializeSecurityExtension() {
```

Comment:
This test won't be necessary once you remove the `initialize` call from the WSQueryEndpoint